src/presentation/http/http-api.ts

```diff
@@ -0,0 +1,234 @@
+   */
+  private static async addAPI(server: FastifyServer, config: HttpApiConfig, domainServices: DomainServices, middlewares: Middlewares): Promise<void> {
+    await server.register(NoteRouter, {
+      prefix: '/note',
+      noteService: domainServices.noteService,
+      middlewares: middlewares,
+    });
+
+    await server.register(OauthRouter, {
+      prefix: '/oauth',
+      userService: domainServices.userService,
+      authService: domainServices.authService,
+      cookieDomain: config.cookieDomain,
+    });
+
+    await server.register(AuthRouter, {
+      prefix: '/auth',
+      authService: domainServices.authService,
+    });
+
+    await server.register(UserRouter, {
+      prefix: '/user',
+      userService: domainServices.userService,
+      middlewares: middlewares,
+    });
+
+    await server.register(AIRouter, {
+      prefix: '/ai',
+      aiService: domainServices.aiService,
+    });
+
+    await server.register(EditorToolsRouter, {
+      prefix: '/editor-tools',
+      editorToolsService: domainServices.editorToolsService,
+    });
+  }
+
+  /**
+   * Registers oauth2 plugin
+   *
+   * @param server - fastify server instance
+   * @param config - http server config
+   */
+  private static async addOauth2(server: FastifyServer, config: HttpApiConfig): Promise<void> {
+    await server.register(fastifyOauth2, {
+      name: 'googleOAuth2',
+      scope: ['profile', 'email'],
+      credentials: {
+        client: {
+          id: config.oauth2.google.clientId,
+          secret: config.oauth2.google.clientSecret,
+        },
+        auth: fastifyOauth2.GOOGLE_CONFIGURATION,
+      },
+      startRedirectPath: config.oauth2.google.redirectUrl,
+      callbackUri: config.oauth2.google.callbackUrl,
+    });
+  }
+
+  /**
+   * Allows cors for allowed origins from config
+   *
+   * @param server - fastify server instance
+   * @param config - http server config
+   */
+  private static async addCORS(server: FastifyServer, config: HttpApiConfig): Promise<void> {
+    await server.register(cors, {
+      origin: config.allowedOrigins,
+    });
+  }
+
+  /**
+   * Add Fastify schema for validation and serialization
+   *
+   * @param server - fastify server instance
+   */
+  private static addSchema(server: FastifyServer): void {
+    server.addSchema(UserSchema);
+  }
+
+  /**
+   * Custom method for sending 404 error
+   *
+   * @param server - fastify server instance
+   */
+  private static add404Handling(server: FastifyServer): void {
+    server.decorate('notFound', NotFoundDecorator);
+  }
+
+
+  /**
+   * Runs http server
+   */
+  public async run(): Promise<void> {
+    if (this.server === undefined || this.config === undefined) {
+      throw new Error('Server is not initialized');
+    }
+    await this.server.listen({
+      host: this.config.host,
+      port: this.config.port,
+    });
+  }
+
+  /**
+   * Returns server instance
+   */
+  public getServer(): FastifyServer | undefined {
```

Comment:
it still returns FastifyServer, so outside-code is still depends on Fastify. Could we get rid of that dependence?

Reply:
Added `fakeRequest()` method to api. It will encapsulate interaction with fastify